### PR TITLE
Finish documenting VisualServer

### LINKED ISSUE #34845: MultiMesh custom data format usage is counterintuitive
Hello, I've been working with the new MultiMeshInstance2D in the beta4 and I noticed something strange about how the custom data format works.
The problem lies in `RasterizerStorageGLES3::multimesh_instance_set_custom_data`.
The docs state that the Color type is only used to hold 4 numbers, however inside the mentioned function, the color is converted from float to 8 bit representation like this
```
data8[0] = CLAMP(p_custom_data.r * 255.0, 0, 255);
data8[1] = CLAMP(p_custom_data.g * 255.0, 0, 255);
data8[2] = CLAMP(p_custom_data.b * 255.0, 0, 255);
data8[3] = CLAMP(p_custom_data.a * 255.0, 0, 255);
```
which is fine for colors, but does in fact change the values of the colors sent to the method.
i.e. I want to send value 2 to the shader, this value lies in the interval 0-255 which I expect to fit in a byte. However, this will not work, because of the transformation which treats the value as if it was a color and clamps it to 255, which will be transformed to 1 inside shader due to how the data attrib pointers are set up. With floats, this works fine, because the numbers are set directly as they are:
```
dataptr[0] = p_custom_data.r;
dataptr[1] = p_custom_data.g;
dataptr[2] = p_custom_data.b;
dataptr[3] = p_custom_data.a;
```
So the number 2 will be usable in the shader as one would expect.
This can be, of course, fixed by sending numbers in the supported float range of 0-1, which would be transformed to the 0-255 range and then multiplying in the shader to get the desired value, but I think such a thing should either not work like this, or be explained in the docs - changing the data type drastically changes how one should work with the API.
I suggest either:
- change the docs and make this obvious to the user
- do not transform the numbers and multiply by 255 in the generated shader, so the values remain the same, limiting usage of "byte" data type to integral numbers (this being made obvious in the docs), while if someone wants to work with floats, they would just pick "float" without having to do any extra computations inside shaders or when calling the functions. Which is, imho a better method of fixing this.

I think this should be discussed and I can make a PR if someone reviews this and agrees on a fix method.

### PR DESCRIPTION
**Pull request by clayjohn**
Fixes: https://github.com/godotengine/godot/issues/34845

This PR finishes the documentation for Visual Server and bumps documentation completion up to 83%

I also include some fixes to MultiMesh to address an open issue.

### CODE REVIEW

**Comment by akien-mga** [comment]
See static checks:
```
ERROR: Unresolved method 'viewport_attach_scenario', file: VisualServer
ERROR: Unresolved member 'ReflectionProbe.boc_projection', file: VisualServer
```

File: doc/classes/VisualServer.xml [hunk 1]
```diff
@@ -6,8 +6,16 @@
 	<description>
 		Server for anything visible. The visual server is the API backend for everything visible. The whole scene system mounts on it to display.
 		The visual server is completely opaque, the internals are entirely implementation specific and cannot be accessed.
+		The visual server can be used to bypass the scene system entirely.
+		Resources are created using the [code]*_create[/code] functions.
+		All objects are drawn to a viewport. You can use the [Viewport] attached to the [SceneTree] or you can create one yourself with [method viewport_create]. When using a custom scenario or canvas, the scenario of canvas needs to be attached to the viewport using [method viewport_attach_scenario] or [method viewport_attach_canvas].
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: Typo: "the scenario **of** canvas"

File: doc/classes/VisualServer.xml [hunk 2]
```diff
@@ -6,8 +6,16 @@
...
+		In 3D, all visual objects must be associated with a scenario. The scenario is a visual representation of the world. If accessing the visual server from a running game, the scenario can be accessed from the scene tree from any [Spatial] node with [method Spatial.get_world]. Otherwise, a scenario can be created with [method scenario_create].
+		Similarly in 2D, a canvas is needed to draw all canvas items.
+		In 3D, all visible objects are comprised of a resource and an instance. A resource can be a mesh, a particle system, a light, or any other 3d object. In order to be visible resources must be attached to instance using [method instance_set_base]. The instance must also be attached to the scenario using [method instance_set_scenario] in order to be visible.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: Typo: "3d" (should be "3D").
"attached to instance" (needs "an")

File: doc/classes/VisualServer.xml [hunk 3]
```diff
@@ -6,8 +6,16 @@
...
+		In 2D all visible objects are some form of canvas item. In order to be visible, a canvas item needs to be the child of a canvas attached to a viewport, or it needs to be the child of another canvas item that is eventually attached to the canvas.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: Add comma after "In 2D" to match previous paragraph.

File: doc/classes/VisualServer.xml [hunk 4]
```diff
@@ -3211,6 +3231,7 @@
 			<argument index="1" name="enable" type="bool">
 			</argument>
 			<description>
+				If [code]true[/code], use box projection. This can make reflections look more correct in certain situations. Equivalent to [member ReflectionProbe.boc_projection].
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: uses

File: doc/classes/VisualServer.xml [hunk 5]
```diff
@@ -4220,12 +4259,16 @@
 			Marks the back side of a cubemap.
 		</constant>
 		<constant name="TEXTURE_TYPE_2D" value="0" enum="TextureType">
+			Normal texture with 2 dimension, width and height.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: dimensions

File: doc/classes/VisualServer.xml [hunk 6]
```diff
@@ -4220,12 +4259,16 @@
...
 		</constant>
 		<constant name="TEXTURE_TYPE_CUBEMAP" value="1" enum="TextureType">
+			Texture made up of six faces, can be looked up with a [code]vec3[/code] in shader.
 		</constant>
 		<constant name="TEXTURE_TYPE_2D_ARRAY" value="2" enum="TextureType">
+			An array of 2 dimensional textures.
 		</constant>
 		<constant name="TEXTURE_TYPE_3D" value="3" enum="TextureType">
+			A 3 dimensional texture with width, height, and depth.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: I think a hyphen might be needed on "3-dimensional" (same above for 2).

File: doc/classes/VisualServer.xml [hunk 7]
```diff
@@ -4414,47 +4461,67 @@
...
+			Proportion of shadow atlas occupied by the third split. The fourth split occupies the rest.
 		</constant>
 		<constant name="LIGHT_PARAM_SHADOW_NORMAL_BIAS" value="12" enum="LightParam">
+			Normal bias used to offset shadow lookup by object normal. Can be used to fix self-shadowing artifacts.
 		</constant>
 		<constant name="LIGHT_PARAM_SHADOW_BIAS" value="13" enum="LightParam">
+			Bias the shadow lookup to fix self-shadowing artifacts.
 		</constant>
 		<constant name="LIGHT_PARAM_SHADOW_BIAS_SPLIT_SCALE" value="14" enum="LightParam">
+			Increases bias on further splits fix self-shadowing that only occurs far away from the camera.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: **to** fix

File: doc/classes/VisualServer.xml [hunk 8]
```diff
@@ -4660,8 +4743,10 @@
 			The amount of vertex memory used.
 		</constant>
 		<constant name="FEATURE_SHADERS" value="0" enum="Features">
+			Hardware supports shaders.
```
Comment (akien-mga) [inline doc/classes/VisualServer.xml]: I think you documented earlier that `has_feature` is not implemented?

**Comment by clayjohn** [comment]
Fixed!

**Review by akien-mga — COMMENTED** [review]
Awesome work!

**Comment by clayjohn** [comment]
Thanks for the feedback!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**